app/test_main.py

```diff
@@ -1,3 +1,74 @@
+import pytest
+
 from app.main import get_human_age
 
-# write your code here
+
+class TestGetHumanAge:
+    @pytest.mark.parametrize(
+        "cat_age, dog_age, expected_age",
+        [
+            (
+                14,
+                14,
+                [0, 0]
+            ),
+            (
+                15, 15, [1, 1]
+            ),
+            (
+                23, 23, [1, 1]
+            ),
+            (
+                24, 24, [2, 2]
+            ),
+            (
+                27, 27, [2, 2]
+            ),
+            (
+                28, 28, [3, 2]
+            ),
+            (
+                100, 100, [21, 17]
+            ),
+            (
+                -5, 1, [0, 0]
+            ),
+            (
+                16.5, 28.1, [1, 2]
+            ),
+            (
+                0, 0, [0, 0]
+            )
+        ]
+    )
+    def test_get_human_age(
+            self, cat_age: int,
+            dog_age: int,
+            expected_age: list
+    ) -> None:
+        assert get_human_age(cat_age, dog_age) == expected_age
+
+    @pytest.mark.parametrize(
+        "cat_age, dog_age, error",
+        [
+            (
+                (),
+                (),
+                TypeError
+            ),
+            (
+                28, (), TypeError
+            ),
+            (
+                (28, 4), (28, 1), TypeError
```

Comment:
Add tests with datatypes other than tuples for this task (e.g. str, dict, set, list, e.g.).